File: agnostic/crates/sv-check/tests/fixtures/semgrep-aisvs/app/assistant.js
// The help-desk assistant again, in JavaScript, with the faults the JavaScript rules look for.
import Anthropic from "@anthropic-ai/sdk";
import { CohereClientV2 } from "cohere-ai";
import express from "express";
import OpenAI from "openai";

const app = express();
const openai = new OpenAI();
const anthropic = new Anthropic();
const cohere = new CohereClientV2();

app.post("/ask", async (req, res) => {
  const reply = await openai.chat.completions.create({
    model: "gpt-5",
    max_tokens: 400,
    messages: [
      { role: "system", content: req.body.persona },
      { role: "user", content: "What are your opening hours?" },
    ],
  });
  res.send(reply.choices[0].message.content);
});

app.post("/summary", async (req, res) => {
  const reply = await anthropic.messages.create({
    model: "claude-sonnet-5",
    system: req.body.instructions,
    messages: [{ role: "user", content: "Summarize the ticket." }],
  });
  res.send(reply.content[0].text);
});

app.post("/triage", async (req, res) => {
  const reply = await cohere.chat({
    model: "command-a",
    safetyMode: "OFF",
    messages: [{ role: "user", content: "Sort this ticket." }],
  });
  res.send(reply.message.content[0].text);
});

File: agnostic/crates/sv-check/tests/fixtures/semgrep-aisvs/app/assistant.py
"""A small help-desk assistant, written with each fault the AISVS rules look for."""
import os

import requests
from flask import Flask, request
from langchain_experimental.utilities import PythonREPL
from mcp.server.fastmcp import FastMCP
from openai import OpenAI

app = Flask(__name__)
client = OpenAI(api_key=os.environ["OPENAI_API_KEY"])
mcp = FastMCP("helpdesk")


@app.post("/ask")
def ask():
    tone = request.form.get("tone")
    reply = client.chat.completions.create(
        model="gpt-5",
        messages=[
            {"role": "system", "content": "You are a help-desk assistant. Tone: " + tone},
            {"role": "user", "content": request.form["question"]},
        ],
    )
    return reply.choices[0].message.content


def keep_trying(question):
    while True:
        reply = client.chat.completions.create(model="gpt-5", max_tokens=400, messages=[
            {"role": "user", "content": question}])
        if reply.choices:
            question = reply.choices[0].message.content


def calculate(expression):
    repl = PythonREPL()
    return repl.run(expression)


@mcp.tool()
def account(name: str) -> dict:
    return {"name": name, "api_key": os.environ["SERVICE_KEY"]}


@mcp.tool()
def weather(city: str) -> str:
    response = requests.get("https://weather.example/api", params={"q": city})
    return response.text
